io/driver/module_unload_load.py

```diff
@@ -0,0 +1,139 @@
+            out = process.getoutput(each.split('/')[-1])
+            if module == out.split('.'[0]):
+                return True
+            return False
+
+    def get_depend_modules(self, module):
+        """
+        Returns the dependent modules
+        """
+        config_path = os.path.join(os.path.abspath(''),
+                                   "module_unload_load.py.data/config")
+        for line in genio.read_all_lines(config_path):
+            if module in line:
+                return line.split('=')[-1]
+
+    def module_load_unload(self, module_list):
+        """
+        Unloading and loading the given module
+        """
+        for mod1 in module_list:
+            if linux_modules.module_is_loaded(mod1) is False:
+                linux_modules.load_module(mod1)
+                time.sleep(self.load_unload_sleep_time)
+
+        for _ in range(0, self.iteration):
+            for mdl in module_list:
+                sub_mod = self.get_depend_modules(mdl)
+                if sub_mod:
+                    sub_mod = sub_mod.split(' ')
+                    for mod in sub_mod:
+                        if mod == 'multipath':
+                            process.system("multipath -F", ignore_status=True)
+                            cmd = "lsmod | grep -i ^%s" % mdl
+                            for _ in range(20):
+                                time.sleep(3)
+                                out = process.getoutput(cmd).split(" ")[-1]
+                                if out.strip() != '0':
+                                    self.log.info("command value : %s\n" % out)
+                                    process.system("multipath -F",
+                                                   ignore_status=True)
+                                    continue
+                                else:
+                                    break
+                        else:
+                            self.log.info("unloading sub module %s " % mod)
+                            process.system("rmmod %s" % mod)
+                self.log.info("unloading module %s " % mdl)
+                process.system("rmmod %s" % mdl)
+                time.sleep(self.load_unload_sleep_time)
+                if linux_modules.module_is_loaded(mdl) is True:
+                    self.error_modules.append(mdl)
+                self.log.info("loading module : %s " % mdl)
+                linux_modules.load_module(mdl)
+                time.sleep(self.load_unload_sleep_time)
+                if linux_modules.module_is_loaded(mdl) is False:
+                    self.error_modules.append(mdl)
+
+    def test(self):
+        """
+        Begining the test here
+        """
+        pci_addrs = []
+        if self.module:
+            for modl in self.module:
+                self.mod_list.append(modl)
+        elif self.only_io is True:
+            pci_addrs = pci.get_pci_addresses()
+            for pci1 in pci_addrs:
```

Comment:
```
elif self.only_io is True: 
    for pci_addr in pci.get_pci_addresses():
        self.mod_list.append(pci.get_driver(pci_addr))
```

Reply:
if we use this, this will end up calling " pci.get_pci_addresses():" each time.
so making it to static. so felt making static if better than calling a function multiple time.

Reply:
we will still call it only once.